### PR TITLE
enable GCP read only creds to be used when root creds missing

### PR DESCRIPTION
**Pull request by joelddiaz**
- [x] add new GCP read-only CredentialsRequest manifests
- [x] implement the ability to use the CCO's read-only GCP creds for verifying that creds need no updates
- [x] actually skip the service API checks when the CredReq says to skip the checking
- [x] add test case to cover the root creds missing but the read-only creds are available
- [x] update readme to mark GCP as supported for removing the root creds
- [x] (unrelated to GCP) return unexpected error in AWS actuator

### CODE REVIEW

File: README.md
Comment (dgoodwin) [inline README.md:203]: This should be 4.7+ rather than Y.

File: pkg/gcp/actuator/actuator.go
Comment (dgoodwin) [inline pkg/gcp/actuator/actuator.go:51]: What was this var doing here already? Looks like there was some half r/o support?

File: pkg/gcp/actuator/actuator.go
Comment (dgoodwin) [inline pkg/gcp/actuator/actuator.go:522]: cllient

File: pkg/operator/credentialsrequest/credentialsrequest_controller_gcp_test.go
Comment (dgoodwin) [inline pkg/operator/credentialsrequest/credentialsrequest_controller_gcp_test.go:379]: Could we get tests for: cred needs a modification but no root cred, cred is missing entirely but no root cred.

File: pkg/gcp/actuator/actuator.go
Comment (dgoodwin) [inline pkg/gcp/actuator/actuator.go:531]: Looks like we also need to set upgradeable=false when that root cred is gone somewhere in the status area. Is it feasible to implement that now in the status handler in this branch, or would we be better waiting until the status controller merges?

**Review by dgoodwin — CHANGES_REQUESTED** [review]

File: pkg/gcp/actuator/actuator.go
Comment (joelddiaz) [reply pkg/gcp/actuator/actuator.go:51]: Yes. When I originally wrote the GCP support, I apparently plumbed just about everything through for read-only support and we were just missing an appropriate role that we could use. There was even an unused buildReadOnlyGCPClient() already written ¯\_(ツ)_/¯

File: pkg/gcp/actuator/actuator.go
Comment (joelddiaz) [reply pkg/gcp/actuator/actuator.go:531]: Good catch! I've put int he upgradeable() implementation to watch for the missing root creds, although one of these PRs might need some changes depending on which one gets in first.

**Comment by dgoodwin** [comment]
Looks good, lets squash and I'll lgtm. Thanks!

File: pkg/gcp/actuator/actuator.go
Comment (joelddiaz) [reply pkg/gcp/actuator/actuator.go:788]: Good question. I just copied what we presently use for the AWS actuator https://github.com/openshift/cloud-credential-operator/blob/master/pkg/aws/actuator/actuator.go#L1251

**Comment by joelddiaz** [comment]
@dgoodwin don't know what it means for you to have outstanding changes requested on the PR, but I think we're good to go minus the question that @akhil-rane brought up about the text we use in the logging.

File: pkg/gcp/actuator/actuator.go
Comment (dgoodwin) [reply pkg/gcp/actuator/actuator.go:788]: I can't remember either. :) We should decide this after a call someday and put it in the readme.

**Pull request merged**
